lightning/src/util/enforcing_trait_impls.rs

```diff
@@ -175,20 +195,18 @@ impl Sign for EnforcingSigner {}
 impl Writeable for EnforcingSigner {
 	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), Error> {
 		self.inner.write(writer)?;
-		let last = *self.last_commitment_number.lock().unwrap();
-		last.write(writer)?;
+		// NOTE - the commitment state is maintained by KeysInterface, so we don't persist it
```

Comment:
Except in the current patch it isnt? It looks like we always generate a fresh `EnforcementState` on every read?

Reply:
There are two ways of reading `EnforcingSigner` from storage in the code - `OnlyReadsKeysInterface::read_chan_signer` and `TestKeysInterface::read_chan_signer`.

`OnlyReadsKeysInterface` is a quick stub implementation for `KeysInterface`.  It is the only one that uses `EnforcingSigner::Readable`.

The "preferred way" is partially outlined in `TestKeysInterface` - where the `InMemorySigner` is read/written individually, but the state is maintained by the `KeysInterface`.  It's not a complete implementation because we don't persist `TestKeysInterface`.

Note that the ldk-sample uses `InMemorySigner`, so there's actually no good illustration of the "preferred way" to handle `EnforcementState`.

So the note is aspirational, maybe it should say "should be maintained".

Reply:
Can we convert tests to use the `new_with_revoked` in `read_chan_signer` of `TestKeysInterface` so that its a bit less aspirational? Dunno how many tests just recreate `TestKeysInterface` willy-nilly, but at least its something? Our enforcing keys signer stuff is mostly just for tests anyway, to ensure that we have additional checking of the channel state machine akin to what a hardware wallet would do, not so much targeting production use-cases, that's what lightning-signer is for :p

Reply:
Take a look at the bottom commit and see if it's in the right direction.